ops/model.py

```diff
@@ -883,11 +991,12 @@ def from_name(cls, name: str, message: str):
             return cls._statuses[name](message)
 
     @classmethod
-    def register(cls, child):
+    def register(cls, child: Type['StatusBase']):
         """Register a Status for the child's name."""
-        if child.name is None:
+        name = typing.cast(Optional[str], child.name)
+        if name is None:
             raise AttributeError('cannot register a Status which has no name')
```

Comment:
If we make the name prop non-abstract and directly have the raise as noted above, then this cast+check also becomes unnecessary.